### PR TITLE
Implement of soft bond potential 

### PR DESCRIPTION
**Pull request by kexul**
Dear Sire Developers: 
    This is an implementation of [soft bond potential ](https://pubs.acs.org/doi/10.1021/acs.jctc.6b00991), including the following changes: 
1. Added a new perturbation type called ScalePerturbation to take care of the exclusion list changes during ring open/close. The exclusion list changes are found by biosimspace, I'll create a PR later.  
2. Decoupled the LJ and coulomb scale parameters from the 14 interactions, so they could be interpolated by lambda. 
3. Implemented the soft bond potential function for bond perturbation. 

I've tested the Fxa series in the paper mentioned above, here is the result: 
|pair  |soft-bond run1       |soft-bond run2     |soft-bond run3        |current |exp  |schrodinger|
|------|-----------|---------|------------|--------|-----|-----------|
|4c->edo|-1.36515025|-1.579566|-1.599211   |-4.11   |-0.8 |-1.48      |
|4d->edo|-1.1339635 |-0.99799 |-1.238629167|-3.29   |-0.87|-1.69      |
|edo->4c|2.0459925  |2.1593732|1.834021    |5.8     |0.8  |1.48       |
|edo->4d|2.04595    |1.2482862|0.824815167 |1.18    |0.87 |1.69       |

From the overlap matrix and pmf plot, we could see that the ring open/close is well treated: 

Overlap matrix of current implementation:
edo~4d_Run_Bound_overlap

PMF of current implementation:
edo~4d_Run_Bound_pmf

Overlap matrix of soft-bond potential:
edo~4d_Run_Bound_overlap_soft_bond

PMF of soft-bond potential: 
edo~4d_Run_Bound_pmf_soft_bond

Looking forward to your comments!

### CODE REVIEW

File: corelib/src/libs/SireMove/openmmfrenergyst.cpp [hunk 1]
```diff
@@ -2932,8 +3538,6 @@ void OpenMMFrEnergyST::createContext(IntegratorWorkspace &workspace, SireUnits::
             integrator_openmm = new OpenMM::VariableVerletIntegrator(integration_tol); //integration tolerance error unitless
         else if (Integrator_type == "langevin")
             integrator_openmm = new OpenMM::LangevinIntegrator(converted_Temperature, converted_friction, dt);
-        else if (Integrator_type == "langevinmiddle")
-            integrator_openmm = new OpenMM::LangevinMiddleIntegrator(converted_Temperature, converted_friction, dt);
```
Comment (lohedges) [inline corelib/src/libs/SireMove/openmmfrenergyst.cpp:2936]: It looks like your merge might have deleted some of the recent additions. Support for the `langevinmiddle` integrator was recently added by @msuruzhon. I'll have check through to see if anything else is missing.

**Comment by jmichel80** [comment]
Hi @kexul many thanks for undertaking this significant piece of work. I will need some time to review carefully the implementation before I comment on the technical approach

**Comment by kexul** [comment]
It's my pleasure to contribute to this great open source project. 😄 
More tests are underway, I'll post updates here.

**Comment by kexul** [comment]
TPSB2 series: 

|pair     |soft-bond         |current imp        |exp                |schrodinger |
|---------|------------------|-------------------|-------------------|------------|
|mol1->mol2|0.74±0.03         |1.15±0.10          |0.62               |0.16        |
|mol2->mol1|-0.81±0.07        |-1.29±0.17         |-0.62              |-0.16       |

**Pull request closed without merging**